src/buffer/out/textBuffer.cpp

```diff
@@ -2358,3 +2362,85 @@ void TextBuffer::CopyHyperlinkMaps(const TextBuffer& other)
     _hyperlinkMap = other._hyperlinkMap;
     _hyperlinkCustomIdMap = other._hyperlinkCustomIdMap;
 }
+
+// Method Description:
+// - Adds a regex pattern we should search for
+// - The searching does not happen here, we only search when asked to by TerminalCore
+// Arguments:
+// - The regex pattern
+// Return value:
+// - An ID that the caller should associate with the given pattern
+const size_t TextBuffer::AddPatternRecognizer(const std::wstring_view regexString)
+{
+    ++_currentPatternId;
+    _IdsAndPatterns.emplace(std::make_pair(_currentPatternId, regexString));
+    return _currentPatternId;
+}
+
+// Method Description:
+// - Copies the patterns the other buffer knows about into this one
+// Arguments:
+// - The other buffer
+void TextBuffer::CopyPatterns(const TextBuffer& OtherBuffer)
+{
+    _IdsAndPatterns = OtherBuffer._IdsAndPatterns;

```

Comment:
i think we need to copy `_currentPatternId` too

Reply:
i just realized we probably needed to copy `_currentHyperlinkId` _too_... but that was from a different PR ;P

(potential bug: resizing may cause us to accidentally recycle hyperlink or pattern IDs)

Reply:
Copied _currentPatternId, will do the hyperlink one in a separate PR